app/src/components/testCaseList/testCaseList.jsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { useIntl } from 'react-intl';
+import { FilterOutlineIcon } from '@reportportal/ui-kit';
+import { SearchField } from 'components/fields/searchField';
+import { withFilter } from 'controllers/filter';
+import { TestCaseCard } from './testCaseCard/testCaseCard';
+import { mockTestCases } from './mockData';
+import { NAMESPACE, SEARCH_KEY, DEFAULT_CURRENT_PAGE } from './testCaseList.constants';
+import { messages } from './messages';
+import styles from './testCaseList.scss';
+
+const cx = classNames.bind(styles);
+
+const SearchFieldWithFilter = withFilter({
+  filterKey: SEARCH_KEY,
+  namespace: NAMESPACE,
+})(SearchField);
+
+export const TestCaseList = ({
+  testCases = mockTestCases,
+  loading = false,
+  currentPage = DEFAULT_CURRENT_PAGE,
+  itemsPerPage,
+  searchValue = '',
+  onSearchChange,
+  onEdit,
+  onDelete,
+  onDuplicate,
+  onMove,
+}) => {
+  const { formatMessage } = useIntl();
+
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+  const currentData = testCases.slice(startIndex, endIndex);
+
+  const handleSearchChange = useCallback(
```

Comment:
useCallback makes sense only if SearchFieldWithFilter is memoised

Reply:
fixed